python/ml4ir/applications/ranking/model/metrics/metrics_helper.py

```diff
@@ -202,7 +207,9 @@ def compute_secondary_labels_metrics_on_query_group(
                 compute_secondary_label_metrics(
                     secondary_label_values=query_group[secondary_label],
                     ranks=query_group[old_rank_col],
-                    click_rank=query_group[query_group[label_col] == 1][old_rank_col].values[0],
+                    click_rank=query_group[query_group[label_col] == 1][old_rank_col].values[0]
+                    if (query_group[label_col] == 1).sum() != 0
+                    else float("inf"),
```

Comment:
`inf` would throw an exception down the path in `compute_secondary_label_metrics` which would be caught in try, catch. I don't see a way around that currently.